scripts/animebytes.coffee

```diff
@@ -61,7 +64,11 @@ module.exports = (robot) ->
 
   robot.hear /^!user(?: (.+))?/i, (msg) ->
     room = msg.message.room
-    name = msg.match[1] || parseUserInfo(msg.message.user.original.host).user
+    try
+      name = msg.match[1] || parseUserInfo(msg.message.user.original.host).user
```

Comment:
Node, as any other language, optimizes execution flow, hence if msg.match[1] satisfies simple '||' expression, it will forgo calling parseUserInfo (which is pointless at this point). Due to that, if unauthorized user calls '!user test', code will never have a chance of throwing exception. Should you change order in this expression, then parseUserInfo will always be true for authorized users and parameters of !user will never work.